scripts/irods/test/test_auth.py

```diff
@@ -252,3 +252,499 @@ def test_allowlist(self):
+        to_return = d.copy()
+
+        for k in other:
+            to_return[k] = other[k]
+
+        return {k: to_return[k] for k in sorted(to_return)} if sort else to_return
+
+
+    def assert_basic_iinit_prompts_are_in_stdout(self, stdout):
+        self.assertIn('Enter the host name (DNS) of the server to connect to', stdout)
+        self.assertIn('Enter the port number', stdout)
+        self.assertIn('Enter your iRODS user name', stdout)
+        self.assertIn('Enter your iRODS zone', stdout)
+
+
+    def assert_ssl_iinit_prompts_are_in_stdout(self, stdout):
+        self.assertIn('Enter the server verification level', stdout)
+        self.assertIn('Enter the full path to the CA certificate file', stdout)
+        self.assertIn('Enter the full path to the certificate key file', stdout)
+        self.assertIn('Enter the full path to the certificate chain file', stdout)
+        self.assertIn('Enter the full path to the DH parameters file', stdout)
+        self.assertIn('Enter the encryption algorithm', stdout)
+        self.assertIn('Enter the encryption key size', stdout)
+        self.assertIn('Enter the encryption salt size', stdout)
+        self.assertIn('Enter the number of hash rounds', stdout)
+
+
+    def assert_auth_scheme_iinit_prompts_are_in_stdout(self, stdout):
+        self.assertIn('Enter your iRODS authentication scheme', stdout)
+
+
+    def assert_environment_file_contents(self, expected_json_contents=None, environment_file_path=None):
+        if expected_json_contents is None:
+            expected_json_contents = self.base_client_environment
+
+        if environment_file_path is None:
+            environment_file_path = self.environment_file_path
+
+        self.assertTrue(os.path.exists(environment_file_path))
+        with open(environment_file_path, 'r') as f:
+            environment_file_contents = json.load(f)
+
+        self.assertEqual(environment_file_contents, expected_json_contents)
+
+
+    @unittest.skipIf(test.settings.USE_SSL, "Basic iinit usage does not configure SSL.")
+    def test_iinit_basic_success(self):
+        # Execute iinit with the basic prompts only, with only correct answers.
+        cmd = 'iinit'
+        user_input = os.linesep.join(
+            [self.host, self.port, self.user, self.zone, self.password]) + os.linesep
+        stdout, stderr, rc = lib.execute_command_permissive(cmd, input=user_input, env=os.environ)
+        lib.log_command_result(cmd, stdout, stderr, rc)
+
+        # Ensure program terminated gracefully.
+        self.assertEqual(rc, 0)
+
+        self.assert_basic_iinit_prompts_are_in_stdout(stdout)
+        self.assertIn('Enter your current iRODS password', stdout)
+
+        # Make sure that the environment file saved with the correct contents.
+        self.assert_environment_file_contents()
+
+        # Make sure that the authentication succeeded.
+        assert_command('ils', 'STDOUT', f'/{self.zone}/home/{self.user}')
```

Comment:
Why `assert_command` instead of `assert_icommand`?

Reply:
`assert_icommand` is a method of an instance of an `IrodsSession` and I wasn't able to use `IrodsSession`s here because we are trying to run `iinit` from scratch. If there's a way to do it, I'm willing to try, but this was much easier to implement.

Reply:
No, what you have is fine. I was just wondering what the motivation was.

Good enough for me.